pkg/resources/mutation/cluster/provisioningcluster.go

```diff
@@ -53,10 +97,257 @@ func (m *Mutator) Admit(request *admission.Request) (*admissionv1.AdmissionRespo
+	}
+	if annotations != nil {
+		secret.Annotations = annotations
+	}
+	if existing == nil {
+		return m.secret.Create(secret)
+	}
+	if !reflect.DeepEqual(existing.Data, secret.Data) {
+		existing.Data = data
+		return m.secret.Update(existing)
+	}
+	return secret, nil
+}
+
+// GetKubeAPIServerArg returns a map representation of the value of kube-apiserver-arg from the cluster's MachineGlobalConfig.
+// An empty map is returned if kube-apiserver-arg is not set in the cluster.
+func GetKubeAPIServerArg(cluster *v1.Cluster) map[string]string {
+	if cluster.Spec.RKEConfig.MachineGlobalConfig.Data != nil {
+		return toMap(cluster.Spec.RKEConfig.MachineGlobalConfig.Data["kube-apiserver-arg"])
+	}
+	return map[string]string{}
+}
+
+func toMap(input interface{}) map[string]string {
+	args := map[string]string{}
+	parsed := convert.ToInterfaceSlice(input)
+	if parsed != nil {
+		for _, arg := range parsed {
+			key, val, found := strings.Cut(convert.ToString(arg), "=")
+			if !found {
+				logrus.Debugf("skipping argument [%s] which does not have right format", arg)
+				continue
+			}
+			args[key] = val
+		}
+	}
+	return args
+}
+
+// SetKubeAPIServerArg uses the provided arg to overwrite the value of kube-apiserver-arg under the cluster's MachineGlobalConfig.
+// If the provided arg is an empty map, SetKubeAPIServerArg removes the existing kube-apiserver-arg from the cluster's MachineGlobalConfig.
+func SetKubeAPIServerArg(arg map[string]string, cluster *v1.Cluster) *v1.Cluster {
+	var parsed []interface{}
+	for key, val := range arg {
+		parsed = append(parsed, interface{}(fmt.Sprintf("%s=%s", key, val)))
+	}
+	if len(parsed) == 0 {
+		delete(cluster.Spec.RKEConfig.MachineGlobalConfig.Data, "kube-apiserver-arg")
+	} else {
+		if cluster.Spec.RKEConfig.MachineGlobalConfig.Data == nil {
+			cluster.Spec.RKEConfig.MachineGlobalConfig.Data = make(map[string]interface{})
+		}
+		cluster.Spec.RKEConfig.MachineGlobalConfig.Data["kube-apiserver-arg"] = parsed
+	}
+	return cluster
+}
+
+// MachineSelectorFileForPSA generates an RKESystemFiles that mounts the secret which contains
+// the generated admission configuration file to the control plane node
+func MachineSelectorFileForPSA(secretName, mountPath string) rkev1.RKEProvisioningFiles {
+	return rkev1.RKEProvisioningFiles{
+		MachineLabelSelector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				ControlPlaneRoleLabel: "true",
+			},
+		},
+		FileSources: []rkev1.ProvisioningFileSource{
+			{
+				Secret: rkev1.K8sObjectFileSource{
+					Name: secretName,
+					Items: []rkev1.KeyToPath{
+						{
+							Key:  SecretKey,
+							Path: mountPath,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// AddMachineSelectorFile adds the provided RKESystemFiles to the cluster's MachineSelectorFiles list.
+// It is a no-op if the provided RKESystemFiles already exist in the cluster.
+func AddMachineSelectorFile(file rkev1.RKEProvisioningFiles, cluster *v1.Cluster) *v1.Cluster {
```

Comment:
Same as the DropMachineSelectorFile. 
```suggestion
func addMachineSelectorFile(file rkev1.RKEProvisioningFiles, cluster *v1.Cluster) {
```

Reply:
fixed